network/manager.go

```diff
@@ -395,9 +395,21 @@ func (nm *networkManager) CreateEndpoint(cli apipaClient, networkID string, epIn
 	if err != nil {
 		return err
 	}
+	// any error after this point should also clean up the endpoint we created above
+	defer func() {
```

Comment:
Curios to know how are we checking if `cni state save failure` is actually the error that triggered this deferred function ? I see there is stateless CNI code as well that tries to update the endpoint state. If that call fails do we want to delete the endpoint in that scenario as well ?

Reply:
Any failure after creating the endpoint should trigger deleting the endpoint (ex: if stateless CNI fails)